docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,84 @@
+
+The native AOT deployment model uses an IL to native compiler. Native AOT apps don't use a Just-In-Time (JIT) compiler when the application runs. Native AOT apps can run in restricted environments where a JIT is not allowed. Native AOT applications target a specific runtime environment, such as Linux x64 or Windows x64, just like publishing a [self-contained app](index.md#publish-self-contained).
+
+There are some limitations in the .NET native AOT deployment model, with the main one being that run-time code generation is not possible. For more information, see [Impact of using the Native AOT deployment](#impact-of-using-the-native-aot-deployment). The support in the .NET 7 release is targeted towards console-type applications.
+
+> [!WARNING]
+> Native AOT is supported in .NET 7 but only a limited number of libraries are fully compatible with native AOT in .NET 7.
+
+## Publish Native AOT - CLI
+
+Publish a native AOT application using the [dotnet publish](../tools/dotnet-publish.md) command.
+
+01. Add `<PublishAot>true</PublishAot>` to your project file.
+
+    This will produce a native AOT app and shows PublishAot compatibility warnings during build.
+
+    ```xml
+    <PropertyGroup>
+        <PublishAot>true</PublishAot>
+    </PropertyGroup>
+    ```
+
+02. Publish the app for for a specific runtime identifier using `dotnet publish -r <RID>`. There are pre-requisites that are needed to be installed for publishing a native AOT application. Some examples are given below for each platform.
+
+    For Windows pre-requisites, install [Visual Studio 2022](https://visualstudio.microsoft.com/vs/community/), including Desktop development with C++ workload. The following example publishes the app for Windows as a native AOT application on a machine with the required pre-requisites installed.
+
+    `dotnet publish -r win-x64 -c Release`
+
+    For Linux pre-requisites, install clang and developer packages for libraries that .NET runtime depends on.
+
+    *Ubuntu (16.04+)*
+
+    ```sh
+    sudo apt-get install clang zlib1g-dev
+    ```
+
+    *Alpine (3.14+)*
+
+    ```sh
+    sudo apk add clang gcc lld musl-dev build-base zlib-dev
+    ```
+
+    The following example publishes the app for Linux as a native AOT application. A native AOT binary produced on Linux machine is only going to work on same or newer Linux version than the build machine Linux version. For example, native AOT binary produced on Ubuntu 20.04 is going to run on Ubuntu 22.04, but it is not going to run on Ubuntu 16.04.
+
+    `dotnet publish -r linux-arm64 -c Release`
+
+The app will be available in the publish directory and will contain all the code needed to run in it, including the coreclr runtime.
```

Comment:
> including the coreclr runtime

I was under the impression that native AOT apps did not include the CoreCLR runtime.

Reply:
It contains a stripped down version, https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/nativeaot.md

Reply:
```suggestion
The app will be available in the publish directory and will contain all the code needed to run in it, including a stripped-down version of the coreclr runtime.
```

Reply:
@LakshanF